pandera/schemas.py

```diff
@@ -2532,3 +2504,31 @@ def convert_uniquesettings(unique: UniqueSettings) -> Union[bool, str]:
             str(unique) + " is not a recognized report_duplicates value"
         )
     return keep_argument
+
+
+def _validate_columns(column_dict: dict[Any, Column]) -> None:
```

Comment:
let's use `Dict` here instead of `dict` for backwards compat... not sure how this is passing CI

Reply:
Actually, I think dict is the preferred way:

dict (and lowercase tuple, list, etc) all work [starting in python 3.9.](https://peps.python.org/pep-0585/#implementation). In fact, per that PEP585, typing.Dict and friends are deprecated for those later versions.

You are right that before python 3.9 we should be getting errors. But the `from __future__ import annotations` directive at the top of the file is saving us: if you have that directive, all python versions 3.7+ (and pandera only supports 3.7+) will not evaluate these type hints at runtime.

So the `__future__` directive saves us for 3.7 and 3.8, and then in 3.9+ we are supposed to use lowercase dict. So we should just transition now, in prep for when 3.7 and 3.8 are sunsetted.

Note that the `__future__` directive only saves us with type hints, not type aliases that need to be evaluated at runtime. eg

```python
Columns = dict[Any, Column]

def _validate_columns(columns: Columns) -> None:
    pass
```

will fail in python 3.7 and 3.8 because `Columns = dict[Any, Column]` is actually evaluated at runtime.

What do you think, can we keep lowercase dict?

Reply:
> So the __future__ directive saves us for 3.7 and 3.8, and then in 3.9+ we are supposed to use lowercase dict.

nice! wasn't aware `__future__` does this

Reply:
Yeah, maybe the more popular use of `__future__` is to make it so you can forward-reference types, but it also has this (and maybe other?) consequences of having delayed evaluation.